docs/introduction.md

```diff
@@ -67,83 +96,207 @@ Flow is based upon the following set of rules:
+
+Conservative execution nodes only process finalized blocks to avoid wasting resources.
+An execution node can be optimistic — they are allowed to work on unfinalized blocks.
+This can, however, lead to waste or resources if a block is abandoned later.
+Also, validity of unfinalized blocks is not guaranteed — execution node may be processing an invalid block.
 
-External clients submit their transactions to collection nodes. Upon receiving well-formed transactions, collection nodes introduce them to the rest of their cluster.
-The collection nodes of one cluster batch the received transactions into collections. Only a hash reference to a collection is submitted to the consensus nodes for inclusion in a block.
+Execution nodes:
+- receive a finalized block
+- execute the transactions
+- update the execution state of the system
 
-Each cluster of collection nodes generates their collections one at a time. Before a new collection is started, the current one has to be closed and sent to the consensus nodes for inclusion in a block.
-The collection nodes' consensus protocol determines when to start/end a collection and which transactions to include in the collection. The result of that consensus is called a _guaranteed collection_.
+During the execution process the following operations occur:
+- each collection of transactions from the finalized block is retrieved from the relevant cluster
+- execution node executes the transactions according to their canonical order and update the execution state of the system
+- an execution receipt is built for the block. An execution receipt is a cryptographic commitment on the execution result of the block
+- the execution receipt is broadcast to the verification nodes and consensus nodes
 
-##### Verification Nodes
+#### Collection retrieval
+
+Blocks being executed contain a number of guaranteed collections. Guaranteed collections only include a hash of the set of transactions, not their full texts.
+Execution nodes retrieve the collection text from the appropriate cluster, and reconstruct the collection hash from the transaction text to verify data consistency.
+Transactions are processed in order.
+
+#### Block Execution
+
+A block is executed when the previous block's execution result is known.
+This previous execution result is the starting point for the execution of the block.
+
+Execution receipt for a block includes execution nodes signature, execution result and [SPoCKs](#specialized-proof-of-confidential-knowledge).
+An execution receipt is a signed execution result.
+The execution receipts can be used to challenge the claims of an execution node when they are shown to be incorrect.
+They are also used to create proofs of the current state of the chain once they are known to be correct.
+
+The execution result of a block includes:
+- block hash
+- hash reference to the previous execution result
+- one or more chunks
+- commitment to the final execution state of the system after executing the block
+
```

Comment:
It would be good to note somewhere that _only_ execution nodes have access to the execution state within the Flow system. And that this is the motivation behind creation of the DPS in the first place: providing an out-of-band scalable gateway to access the Flow execution state, without putting load on the Flow system itself.